Cleanup testing of _like operators

VitalyFedyunin opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #28544 Add torch.memory_format support to the TorchScript
* #28292 Add memory format support to the `resize_` op.
* #28291 TensroIterator preserve format for binary, ternary operators.
* #28077 Add channels last support to cuda.comm.scatter and gather
* #28076 Kill `operator==` of TensorOptions as confusing one
* #27979 Add memory format support to `resize_as_` operator
* **#27891 Cleanup testing of _like operators**
* #27890 Add memory format support to `randn_like` operator
* #27889 Add memory format support to `randint_like` operator
* #27562 Add memory format support to `zeros_like` operator
* #27561 Add memory format support to `rand_like` operator
* #27270 Add memory format support to `ones_like` operator
* #27262 Add memory format support to `full_like` operator
* #27244 Add memory format support to `empty_like` operator

Differential Revision: [D17980308](https://our.internmc.facebook.com/intern/diff/D17980308)

VitalyFedyunin committed:
Cleanup testing of _like operators

[ghstack-poisoned]

VitalyFedyunin committed:
Fix merge conflict on "Cleanup testing of _like operators"




[ghstack-poisoned]

VitalyFedyunin committed:
Fix .to copy=True path on "Cleanup testing of _like operators"




[ghstack-poisoned]

VitalyFedyunin committed:
Rebase on top of master on "Cleanup testing of _like operators"




[ghstack-poisoned]

VitalyFedyunin committed:
Fixing casting ops and tests on "Cleanup testing of _like operators"




[ghstack-poisoned]

VitalyFedyunin committed:
Rebasing to master on "Cleanup testing of _like operators"




[ghstack-poisoned]

VitalyFedyunin committed:
Rebasing to master on "Cleanup testing of _like operators"


Differential Revision: [D17980308](https://our.internmc.facebook.com/intern/diff/D17980308)

[ghstack-poisoned]

gchanan reviewed (commented):
are you planning to do JIT tests?  or is that coming later?

VitalyFedyunin committed:
Fix rand_like test on "Cleanup testing of _like operators"


Differential Revision: [D17980308](https://our.internmc.facebook.com/intern/diff/D17980308)

[ghstack-poisoned]

VitalyFedyunin committed:
Fix resize_as_ linter on "Cleanup testing of _like operators"


Differential Revision: [D17980308](https://our.internmc.facebook.com/intern/diff/D17980308)

[ghstack-poisoned]

VitalyFedyunin committed:
Fix onnx on "Cleanup testing of _like operators"


Differential Revision: [D17980308](https://our.internmc.facebook.com/intern/diff/D17980308)

[ghstack-poisoned]

VitalyFedyunin committed:
Rebase on "Cleanup testing of _like operators"


Differential Revision: [D17980308](https://our.internmc.facebook.com/intern/diff/D17980308)

[ghstack-poisoned]

VitalyFedyunin committed:
randn_like not simple mapable any more on "Cleanup testing of _like operators"


Differential Revision: [D17980308](https://our.internmc.facebook.com/intern/diff/D17980308)

[ghstack-poisoned]

VitalyFedyunin committed:
randn_like mappable again and supported in jit with memory format on "Cleanup testing of _like operators"


Differential Revision: [D17980308](https://our.internmc.facebook.com/intern/diff/D17980308)

[ghstack-poisoned]

VitalyFedyunin committed:
fix bc test for empty_like on "Cleanup testing of _like operators"


Differential Revision: [D17980308](https://our.internmc.facebook.com/intern/diff/D17980308)

[ghstack-poisoned]

VitalyFedyunin committed:
fix bc test for empty_like now with gradient on "Cleanup testing of _like operators"


Differential Revision: [D17980308](https://our.internmc.facebook.com/intern/diff/D17980308)

[ghstack-poisoned]
